tests/core/pyspec/eth2spec/test/merge/unittests/test_validate_merge_block.py

```diff
@@ -0,0 +1,153 @@
+from typing import Optional
+from eth2spec.utils.ssz.ssz_typing import uint256, Bytes32
+from eth2spec.test.helpers.block import (
+    build_empty_block_for_next_slot,
+)
+from eth2spec.test.helpers.pow_block import (
+    prepare_random_pow_chain,
+)
+from eth2spec.test.context import (
+    spec_state_test,
+    with_merge_and_later,
+    spec_configured_state_test
+)
+
+
+TERMINAL_BLOCK_HASH_CONFIG_VAR = '0x0000000000000000000000000000000000000000000000000000000000000001'
+TERMINAL_BLOCK_HASH = Bytes32(TERMINAL_BLOCK_HASH_CONFIG_VAR)
+
+
+def run_validate_merge_block(spec, pow_chain, beacon_block, valid=True):
+    """
+    Run ``validate_merge_block``
+    If ``valid == False``, run expecting ``AssertionError``
+    """
+
+    def get_pow_block(hash: spec.Bytes32) -> Optional[spec.PowBlock]:
+        for block in pow_chain:
+            if block.block_hash == hash:
+                return block
+        return None
+
+    get_pow_block_backup = spec.get_pow_block
+
+    # Guido authorized everyone to do this
+    spec.get_pow_block = get_pow_block
+    assertion_error_caught = False
+    try:
+        spec.validate_merge_block(beacon_block)
+    except AssertionError:
+        assertion_error_caught = True
+    except Exception as e:
+        spec.get_pow_block = get_pow_block_backup
+        raise e
+    spec.get_pow_block = get_pow_block_backup
+
+    if valid:
+        assert not assertion_error_caught
+    else:
+        assert assertion_error_caught
+
+
+@with_merge_and_later
+@spec_state_test
+def test_validate_merge_block_success(spec, state):
+    pow_chain = prepare_random_pow_chain(spec, 2)
+    pow_chain[0].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY - uint256(1)
+    pow_chain[1].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY
```

Comment:
For this and following this kind of test setup could look more readable I think
```suggestion
    pow_chain[-1].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY
    pow_chain[-2].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY - uint256(1)
```

Reply:
An improve on that could be:
```python
pow_chain.head()
pow_chain.head(-1)
```